pex/pip.py

```diff
@@ -729,33 +751,93 @@ def _fixup_install(
-        ) as record_fi:
-            for line in record_fi:
-                if line.split(",")[0] != direct_url_relpath:
+        with closing(fileinput.input(files=[record_abspath], inplace=True, mode="rU")) as record_fi:
+            csv_writer = None  # type: Optional[CSVWriter]
+            for path, existing_hash, existing_size in csv.reader(
+                record_fi, delimiter=",", quotechar='"'
+            ):
+                if csv_writer is None:
+                    # N.B.: The raw input lines include a newline that varies between '\r\n' and
+                    # '\n' when the wheel was built from an sdist by Pip depending on whether the
+                    # interpreter used was Python 2 or Python 3 respectively. As such, we normalize
+                    # all RECORD files to use '\n' regardless of interpreter.
+                    csv_writer = cast(
+                        "CSVWriter",
+                        csv.writer(sys.stdout, delimiter=",", quotechar='"', lineterminator="\n"),
+                    )
+
+                abspath_to_rehash = to_rehash.pop(path, None)
+                if existing_hash and abspath_to_rehash is not None:
+                    algorithm = existing_hash.split("=")[0]
+                    hasher = hashlib.new(algorithm)
+                    with open(abspath_to_rehash, "rb") as rehash_fp:
+                        CacheHelper.update_hash(rehash_fp, digest=hasher)
+
+                    fingerprint = base64.urlsafe_b64encode(hasher.digest()).decode("ascii")
+                    de_padded, pad, rest = fingerprint.rpartition("=")
+                    new_hash = str(de_padded if pad and not rest else fingerprint)
+                    new_size = os.stat(abspath_to_rehash).st_size
+                    csv_writer.writerow((path, new_hash, new_size))
+                elif path != direct_url_relpath:
+                    csv_writer.writerow((path, existing_hash, existing_size))
+
+    @staticmethod
+    def _fixup_scripts(
+        install_dir,  # type: str
+    ):
+        # type: (...) -> Iterator[str]
+        bin_dir = os.path.join(install_dir, "bin")
+        if not os.path.isdir(bin_dir):
+            return
+
+        scripts = []
+        for script_name in os.listdir(bin_dir):
+            script_path = os.path.join(bin_dir, script_name)
+            if is_exe(script_path):
+                scripts.append(script_path)
+
+        with closing(fileinput.input(files=scripts, inplace=True)) as script_fi:
+            for line in script_fi:
+                if script_fi.isfirstline() and re.match(r"^#!.*python", line):
+                    # Ensure python shebangs are reproducible. The only place these can be used is
+                    # in venv mode PEXes where the `#!python` placeholder shebang will be re-written
+                    # to use the venv's python interpreter.
+                    print("#!python")
```

Comment:
Is this `print()` intentional?

Reply:
Yes. `fileinput.input(..., replace=True)` has an interesting API.